lstchain/high_level/interpolate.py

```diff
@@ -0,0 +1,577 @@
+    irf_unique, irf_num = np.unique(irf_params_short, axis=0, return_counts=True)
+    idx = np.flatnonzero((irf_unique == irf_params_short[index]).all(1))
+
+    if irf_num[idx] > 1:
+        # Only using the overlapping nodes
+        idx_list = np.flatnonzero((irf_params_short == irf_params_short[index]).all(1))
+        # Finding the shortest distance with respect to target azimuth
+        diff = np.abs(irf_params_full[idx_list, 2] - target_params_full[2])
+        idx = idx_list[np.where(diff == diff.min())[0]][0]
+    else:
+        # if there are no overlapping nodes
+        idx = index
+    return idx
+
+
+def compare_irfs(irfs):
+    """
+    Compare the given list of IRFs with data binning and relevant
+    metadata values.
+
+    Parameters
+    ----------
+    irfs: List of IRFs to compare
+        'List'
+    Returns
+    -------
+    : Boolean indicating whether the given list of IRFs are comparable
+    """
+    bin_bool = False
+    meta_bool = True
+
+    params = []
+
+    # Collect the list of metadata and column names to compare the values/data
+    select_meta = ["HDUCLAS3", "INSTRUME", "G_OFFSET"]
+    try:
+        h = Table.read(irfs[0], hdu="EFFECTIVE AREA")
+        h_meta = h.meta
+        cols = h.columns[:-1]
+
+        energy_dependent_gh = "GH_EFF" in h_meta
+        point_like = h_meta["HDUCLAS3"] == "POINT-LIKE"
+        energy_dependent_theta = "TH_CONT" in h_meta
+        source_dep = "AL_CUT" in h_meta
+
+        if energy_dependent_gh:
+            select_meta.append("GH_EFF")
+        else:
+            select_meta.append("GH_CUT")
+
+        if point_like:
+            if energy_dependent_theta:
+                select_meta.append("TH_CONT")
+            else:
+                if source_dep:
+                    select_meta.append("AL_CUT")
+                else:
+                    select_meta.append("RAD_MAX")
+    except KeyError:
+        print(f"Effective Area is not present in {irfs[0]}")
+
+    # Comparing the metadata information
+    for k in select_meta:
+        meta = [Table.read(i, hdu="ENERGY DISPERSION").meta[k] for i in irfs]
+        m = np.unique(meta)
+
+        if len(m) != 1:
+            meta_bool *= False
+
+    print(f"The metadata are{' ' if meta_bool else ' not '}comparable")
+    for irf in irfs:
+        e_table = Table.read(irf, hdu="ENERGY DISPERSION")
+
+        for col in cols:
+            params.append(e_table[col].quantity[0])
+
+    # Comparing other paramater columns in IRFs
+    for i in np.arange(len(cols)):
+        a = [params[len(cols) * j + i] for j in np.arange(len(irfs))]
+        a2, a_ind = np.unique(a, return_index=True)
+
+        if len(a2) == len(params[i]):
+            if (a2[np.argsort(a_ind)] == params[i].value).all():
+                bin_bool = True
+    print(
+        "The other parameter axes data " f"are{' ' if bin_bool else ' not '}comparable"
+    )
+
+    return bin_bool * meta_bool
```

Comment:
While this likely works, I am not sure the multiplication of booleans is common. Someone more python-wise could comment on this.

Reply:
I think, we can save this issue to be resolved later when we try to improve the overall code quality.